cmd/thanos/sidecar.go

```diff
@@ -294,6 +299,23 @@ func runSidecar(
 	return nil
 }
 
+func checkFlags(ctx context.Context, logger log.Logger, promURL *url.URL, tsdbPath string) error {
+	flags, err := promclient.ConfiguredFlags(ctx, logger, promURL)
```

Comment:
We don't want break compatiblity with Prometheuses pre 2.2.1 I think. This means that we need to filter out somehow errors (probably `NotFound`?) from Prometheus and produce warning that `we are unable to validate Prometheus that is older than 2.2.1` (: 

What do you think?